flocker/control/test/test_httpapi.py

```diff
@@ -450,3 +478,299 @@ def test_listens_endpoint(self):
         port = server[0]
         factory = server[1].__class__
         self.assertEqual((port, factory), (6789, Site))
+
+
+class DatasetsStateTestsMixin(APITestsMixin):
+    """
+    Tests for the service datasets state description endpoint at
+    ``/state/datasets``.
+    """
+    def test_empty(self):
+        """
+        Test case where cluster contains no datasets.
```

Comment:
Test method docstrings should explain both what the case is and what the expected behavior is.  This way the intent is captured (at least, hopefully) at least in the docstring - even if the implementation ends up being confusing, misleading, or just plain buggy.  If the intent is known the test can be maintained, if it isn't things get hard.  For this test method I think I'd write something like:

```
When the cluster state includes no datasets, the endpoint returns an empty list.
```